File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int l=0,r=nums.size()-1,Min=INT_MAX;
        while(l<=r){
           int mid=(l+r)/2;
           if(nums[l]<=nums[mid]){
            Min=min(nums[l],Min);
            l=mid+1;
           }
           else{
            Min=min(nums[mid],Min);
            r=mid-1;
           }
            }
        return Min;
        }    
            
    };

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
     int l=0,h=nums.length-1,Min=Integer.MAX_VALUE;
     while(l<=h){
        int mid=l+(h-l)/2;
        if(nums[l]<=nums[mid]){
            Min=Math.min(nums[l],Min);
            l=mid+1;
        }
        else{
            Min=Math.min(nums[mid],Min);
            h=mid-1;
        }
     }   
        return Min;
    }
}